### PR TITLE
iOS Plugin documentation

### PR DESCRIPTION
**Pull request by naithar**
Related PRs for Godot: `master` https://github.com/godotengine/godot/pull/41230, `3.2` https://github.com/godotengine/godot/pull/41340
Proposal: https://github.com/godotengine/godot-proposals/issues/1185

Adds a starting documentation for iOS plugin system implemented in linked PRs.
I've used Android plugin documentation as an example.

Header files could probably be made available just like Android's plugin `.aar`.

### CODE REVIEW

File: tutorials/plugins/ios/index.rst
Comment (Calinou) [inline tutorials/plugins/ios/index.rst:2]: The number of characters below a heading *must* match the heading above:

```suggestion
iOS plugins
===========
```

This is a Sphinx/reStructuredText requirement. Make sure to do this for all the headings you've added.

**Review by Calinou — CHANGES_REQUESTED** [review]

**Comment by NathanLovato** [comment]
Related to #4163 

Thanks for the contribution. Would you mind if I directly edited the document to make the style fit our writing guidelines? https://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html

You can check the guidelines and do it yourself if you'd like. I'm offering to edit for you because it's much more efficient than leaving you dozens of review comments.

**Comment by naithar** [comment]
> Would you mind if I directly edited the document to make the style fit our writing guidelines?

If you think it's more efficient, sure, I wouldn't mind.

**Comment by NathanLovato** [comment]
I opened a new PR with some WIP edits. I hadn't noticed the feature wasn't merged in the engine; we should review and merge the docs only after a feature was added to Godot.

**Comment by naithar** [comment]
> I opened a new PR with some WIP edits. I hadn't noticed the feature wasn't merged in the engine; we should review and merge the docs only after a feature was added to Godot.

Sorry, should have made it more clear in OP.

And it's also strange that you couldn't push directly to this PR as I have enabled `edit by maintainers`

**Pull request closed without merging**
